opal/mca/pmix/base/pmix_base_fns.c

```diff
@@ -193,20 +193,21 @@ static void translate_params(void)
     home = opal_home_directory();
     if (NULL != home) {
         file = opal_os_path(false, home, ".openmpi", "mca-params.conf", NULL);
-        OBJ_CONSTRUCT(&params, opal_list_t);
-        mca_base_parse_paramfile(file, &params);
-        free(file);
-        OPAL_LIST_FOREACH (fv, &params, mca_base_var_file_value_t) {
-            pmix_overlap = check_pmix_overlap(&e2[len], evar);
-            if (!pmix_overlap && check_pmix_param(fv->mbvfv_var)) {
-                opal_asprintf(&tmp, "PMIX_MCA_%s", fv->mbvfv_var);
-                // set it, but don't overwrite if they already
-                // have a value in our environment
-                setenv(tmp, fv->mbvfv_value, false);
-                free(tmp);
+        if (NULL != file) {
+            OBJ_CONSTRUCT(&params, opal_list_t);
+            mca_base_parse_paramfile(file, &params);
+            free(file);
+            OPAL_LIST_FOREACH (fv, &params, mca_base_var_file_value_t) {
+               if (check_pmix_param(fv->mbvfv_var)) {
```

Comment:
This still isn't quite correct. You do need to check the params from the file for overlap - i.e., legacy ORTE params that need to be converted to PMIx. The problem was that we were passing the wrong things to `check_pmix_param` - the params in that function are (the variable name, the variable's value). So here you would pass `check_pmix_overlap(fv->mbvfv_var, fv->mbvfv_value)`.

You still need `check_pmix_param` in case they put a PMIx var in the file - you just need to check overlap as well.

HTH

Reply:
LGTM!